File: src/main/java/com/pubnub/api/UserId.java
package com.pubnub.api;

import lombok.Data;
import org.jetbrains.annotations.NotNull;

import static com.pubnub.api.builder.PubNubErrorBuilder.PNERROBJ_USERID_NULL_OR_EMPTY;

@Data
public class UserId {
    private final String value;

    public UserId(@NotNull String value) {
        PubNubUtil.require(value != null && !value.isEmpty(), PNERROBJ_USERID_NULL_OR_EMPTY);
        this.value = value;
    }
}

File: src/main/kotlin/com/pubnub/api/UserId.kt
package com.pubnub.api

import com.google.gson.annotations.JsonAdapter
import com.pubnub.api.PNConfiguration.Companion.isValid
import com.pubnub.api.utils.DataClassAsValue

@JsonAdapter(DataClassAsValue::class)
data class UserId( val value: String) {
    init {
        PubNubUtil.require(value.isValid(), PubNubError.USERID_NULL_OR_EMPTY)
    }
}
